File: Core-backend-layer/Programming-fundamentals/Java/OOP/src/AbstractClasses/products/ElectronicsProduct.java
package products;

public class ElectronicsProduct extends Product {
    String brand;

    public ElectronicsProduct(int productId, String name, double price, double costPrice, String brand) {
        super(productId, name, costPrice);
        this.price = price; // not initialized in the constructor of the abstract class
        this.brand = brand;
    }

    // Implementation of abstract method
    @Override
    public void displayInfo() {
        System.out.println("Electronics product: " + name + ", Price: $" + price +
                ", Brand: " + brand + ", Tax: $" + calculateTax() +
                ", Profit: $" + calculateProfit());
    }
}

File: Core-backend-layer/Programming-fundamentals/Python/OOP/src/abstract_classes/products/electronics_product.py
from products.product import Product

class ElectronicsProduct(Product):
    def __init__(self, product_id, name, price, cost_price, brand):
        super().__init__(product_id, name, cost_price)
        self.price = price  # initialized in the constructor of the abstract class to 0
        self.brand = brand

    # Implementation of abstract method
    def display_info(self):
        print(f"Electronics product: {self.name}, Price: ${self.price}, "
              f"Brand: {self.brand}, Tax: ${self.calculate_tax()}, "
              f"Profit: ${self._calculate_profit()}")